var/spack/repos/builtin/packages/flux-sched/package.py

```diff
@@ -191,6 +192,7 @@ def setup_run_environment(self, env):
         env.prepend_path("FLUX_MODULE_PATH", self.prefix.lib64.flux.modules.sched)
         env.prepend_path("FLUX_EXEC_PATH", self.prefix.libexec.flux.cmd)
         env.prepend_path("FLUX_RC_EXTRA", self.prefix.etc.flux)
+        env.prepend_path("PYTHON", os.path.join(self.spec["python"].prefix.bin, "python3"))
```

Comment:
Does `setup_run_environment` get called during build such that this is set during cmake invocation? I thought this only got called for run and deps.  Setting this env var would work, as would adding `Python_EXECUTABLE` as a cmake define.

Reply:
You make a good point, and yet I think that is the function I modified for the user which worked.  I'll verify that tomorrow.

Reply:
Yep, I made the change in the same function for the user and it fixed the issue.  I could try relocating (or just copying) it to `setup_build_environment()` if that would make us feel better.

Reply:
Thinking about it more, I would prefer that but mainly so that the PYTHON env var doesn't get set in the user's run environment. We try not to mess with whatever python the user is trying to run.  Sorry for the extra step but I think we'll end up in a better place that way.

Reply:
I don't have an issue with modifying that, but I also don't have any way to test if this will fix the error that was observed.  If we're okay with that, then so be it.